### PR TITLE
 Update stomptemplate.py to check connection existing before disconnecting

### LINKED ISSUE #401: Stomp Tests Timeout Forever
If I run the stomp tests without a valid message bus running, they should fail within the 10 second timeout (they usually fail earlier when disconnection occurs). They currently hang forever, this behaviour seems to have been present since #345 so probably has something to do with the fact that we are now logging exceptions rather than raising them. This would be unfortunate as it means our solution that made the CLI easier to use makes the tests harder to use.

This ticket is therefore to come up with a solution to #220 that also works for this.

### PR DESCRIPTION
**Pull request by stan-dot**
fixes #401

### CODE REVIEW

File: tests/worker/test_reworker.py [hunk 1]
```diff
@@ -380,15 +380,15 @@ def assert_running_count_plan_produces_ordered_worker_and_data_events(
 def take_n_events(
     stream: EventStream[E, Any],
     num: int,
-) -> "Future[list[E]]":
+) -> Future[list[E]]:
```
Comment (callumforrester) [inline tests/worker/test_reworker.py:383]: @DiamondJoseph was this there for a reason? I forget

**Review by callumforrester — CHANGES_REQUESTED** [review]
Please give this PR a better title

File: tests/worker/test_reworker.py (hunk 1 above)
Comment (DiamondJoseph) [reply tests/worker/test_reworker.py:383]: Them being quoted? None that I can recall. If it's working without then great, but I worry given the forward ref issues we had in blueapi recently (but no forward refs here, causality is maintained).

Comment (stan-dot) [reply tests/worker/test_reworker.py:383]: I had to change that otherwise the pre-commit wouldn't pass

**Pull request closed without merging**